### ISSUE TITLE
calling TrackJS.track outputs warning "TrackJS: agent must be installed"

### ISSUE THREAD

**Issue by jusurb** [TrackJs/trackjs-package#32]
After upgrading to v3 whenever we call TrackJS.track() in our code it outputs a warning 
TrackJS: agent must be installed t.js:55

we first load trackjs agent lib and then call TrackJS.install but something isn't entirely right

**Comment by BrandesEric** [comment]
@Gamesh could you include a code snippet of how you're initializing everything?  One thing to be aware of is that, if you're using ES6 `import` - all import statements will be executed at the beginning of the file, regardless of where they appear in the document (at least Babel uses this behavior).

Additionally, are any of your imports using TrackJS.track() before install is called? For example:
![image](https://user-images.githubusercontent.com/640877/48066625-b94be100-e193-11e8-82c4-1aba0fcb4f90.png)

In this scenario, the App component cannot use TrackJS.track() outside of a class method or function call, as it's loaded and executed before TrackJS.install() is called.  

A better way to ensure that you can safely call TrackJS.track() whenever you want is to create a purpose built install file, like the following:
![image](https://user-images.githubusercontent.com/640877/48066760-0f208900-e194-11e8-95ab-48d0b81f89a5.png)

Anyways, if you could provide a quick code snippet that would be very helpful!

Thanks!
Eric

**Comment by jusurb** [comment]
we load TracksJs from cdn first then we call the TrackJS.install

all the vendors that use trackJs.track command are loaded after the inline script that calls install. But even if i call track() from browser console it still prints out the warning and at that point trackjs.install() was already called.

`<script src="https://cdn.trackjs.com/agent/3.0.0/t.js" crossorigin="anonymous"></script>`
```
window.TrackJS && TrackJS.install({
    token: '<...token..>',
    application: 'dev',
    enabled: false,
    cookie: false,
    network: {
        error: false
    },
            userId: '....',
        onError: function (payload) {
                window.ErrorLog.push(payload.message);

        return !window.suppressErrorReporting;
    }
});
```
<img width="488" alt="image" src="https://user-images.githubusercontent.com/171747/49006427-c136e980-f171-11e8-8dca-4ff0646a6718.png">
to be honest i'm not sure it was needed to change from async init like google analytics does by registering a global var to this model where the cdn script already needs to be loaded to call install

**Comment by jusurb** [comment]
ah it seams the warning is printed if enabled is set to false, which it didn't before

**Comment by toddhgardner** [comment]
@Gamesh, we made the change to support explicit initialization and module loading, but we fully support the old way too. If you prefer to setup a `window._trackJs` variable, the script will detect it and automatically install.

The warning is a good point, if you've explicitly disabled the agent, we shouldn't print the warning. Will fix with the next release.

**Comment by toddhgardner** [comment]
This has been fixed in 3.1.0, which has gone to Canary testing today. Will be generally available this week.

**Comment by jusurb** [comment]
thanks
